tests/daemon/test-persistent-cache.c

```diff
@@ -349,1064 +302,449 @@ boot_timestamp_is_valid (gint64 previous_relative_time,
+static guint64
+assert_variants_read (EmerPersistentCache *cache,
+                      GVariant           **variants,
+                      gsize                num_variants)
 {
-  SingularEvent *singular_array;
-  AggregateEvent *aggregate_array;
-  SequenceEvent *sequence_array;
-
-  make_many_events (&singular_array,
-                    &aggregate_array,
-                    &sequence_array,
-                    num_singulars_made,
-                    num_aggregates_made,
-                    num_sequences_made);
-
-  gboolean success = emer_persistent_cache_store_metrics (cache,
-                                                          singular_array,
-                                                          aggregate_array,
-                                                          sequence_array,
-                                                          *num_singulars_made,
-                                                          *num_aggregates_made,
-                                                          *num_sequences_made,
-                                                          num_singulars_stored,
-                                                          num_aggregates_stored,
-                                                          num_sequences_stored,
-                                                          capacity);
-
-  g_free (singular_array);
-  g_free (aggregate_array);
-  for (gint i = 0; i < *num_sequences_made; i++)
-    g_free (sequence_array[i].event_values);
-  g_free (sequence_array);
-
-  return success;
-}
+  GVariant **variants_read;
 
-static void
-assert_singulars_equal_variants (SingularEvent *singular_array,
-                                 gint           singular_array_length,
-                                 GVariant     **variants)
-{
-  for (gint i = 0; i < singular_array_length || variants[i] != NULL; i++)
-    {
-      g_assert_true (i < singular_array_length && variants[i] != NULL);
+  gsize total_elem_size = 0;
+  for (gsize i = 0; i < num_variants; i++)
+    total_elem_size += get_size_with_type (variants[i]);
 
-      GVariant *singular_variant = singular_to_variant (singular_array + i);
-      g_assert_true (g_variant_equal (singular_variant, variants[i]));
-      g_variant_unref (singular_variant);
-    }
-}
+  gsize total_size_when_stored = 0;
+  for (gsize i = 0; i < num_variants; i++)
+    total_size_when_stored += get_size_when_stored (variants[i]);
 
-static void
-assert_aggregates_equal_variants (AggregateEvent *aggregate_array,
-                                  gint            aggregate_array_length,
-                                  GVariant      **variants)
-{
-  for (gint i = 0; i < aggregate_array_length || variants[i] != NULL; i++)
-    {
-      g_assert_true (i < aggregate_array_length && variants[i] != NULL);
+  /* These arbitrary values should be overwritten. */
+  gsize num_variants_read = num_variants + 1;
+  guint64 token = total_size_when_stored + 1;
```

Comment:
It makes the tests more deterministic. In particular, zero is the most common value in uninitialized memory, and often these values are supposed to be zero.